File: swea/D2/1959.두 개의 숫자열/1959.py
T = int(input())
for tc in range(1, T+1):
    n, m = map(int, input().split())
    arr1 = list(map(int, input().split()))
    arr2 = list(map(int, input().split()))

    if len(arr2) > len(arr1):
        arr1[:], arr2[:] = arr2[:], arr1[:]
    
    ans = 0
    for i in range(len(arr1) - len(arr2) + 1):
        total = 0
        for j in range(len(arr2)):
            total += arr1[j + i] * arr2[j]
        
        ans = max(ans, total)
    
    print(f'#{tc} {ans}')

File: swea/D2/1959.두 개의 숫자열/1959.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int T; 

int main(){
  cin >> T;

  for(int t = 1; t <= T; t++){
    int n, m;
    cin >> n >> m;
    
    vector<int> arr1(n);
    vector<int> arr2(m);

    for(int i = 0; i < n; i++) cin >> arr1[i];
    for(int i = 0; i < m; i++) cin >> arr2[i];

    if(m > n){
      swap(n, m);
      swap(arr1, arr2);
    }

    int ans = 0;
    for(int i = 0; i <= n - m; i++){
      int total = 0;
      for(int j = 0; j < m; j++){
        total += arr1[i+j] * arr2[j];
      }

      ans = max(ans, total);
    }

    cout << '#' << t << ' ' << ans << endl;
  }
}